File: hw1/gcd_full.rb
#! /usr/bin/env ruby

def gcdI(i, j)
    while i!= j
        if i > j
            i = i - j
        else
            j = j - i;
        end
    end
    return i
end

def gcdF(l, j)
  if l > j
    if l%j==0
      return j
    else
      return gcdF(l % j,j)
    end
  else
    if j%l==0
      return l
    else
      return gcdF(l,j%l)
    end
  end
  return l
end


if ARGV.length != 2
  puts "fact.rb usage: [NUMBER]"
  exit
end

puts gcdI(ARGV[0].to_i,ARGV[1].to_i)
puts gcdF(ARGV[0].to_i,ARGV[1].to_i)

File: hw1/gcd_full.py
#! /usr/bin/env python
import sys

def gcdI(i, j):
    while (i!= j):
        if (i > j):
            i = i - j
        else:
            j = j - i;
    return i;


def gcdF(i, j):
    while (i!= j):
        if (i > j):
            if(i%j==0):
                return j
            else:
                return gcdF(i % j,j)
        else:
            if(j%i==0):
                return i
            else:
                return gcdF(i,j%i)
    return i;

if len(sys.argv) != 3:
  print("%s usage: [NUMBER]" % sys.argv[0])
  exit()

print(gcdI(int(sys.argv[1]),int(sys.argv[2])))
print(gcdF(int(sys.argv[1]),int(sys.argv[2])))
